firecrown/parser.py

```diff
@@ -1,30 +1,56 @@
 import importlib
 import yaml
 import jinja2
+import collections
 
 
-def parse(filename):
+def parse(source, settings=None):
     """Parse a configuration file.
 
     Parameters
     ----------
-    filename : str
+    source : str
         The config file to parse. Should be YAML formatted.
+        if source has 'read' attribute, it is used to fetch data,
+        otherwise it is interpreted as a filename
+    settings : dictionary, optional
+         Dictionary (can be of dictionaries) used to update parsed
+         config file after reading it. If items in config exist, then
+         they are overwritten, otherwise added. It does not delete existing
+         keys in the config.
 
     Returns
     -------
-    config: dict
-        The raw config file as a dictionary.
+    config : dict
+        The raw config file as a dictionary, updated with settings
     data : dict
         A dictionary containg each analyses key replaced with its
         corresponding data and function to compute the log-likelihood.
     """
 
-    with open(filename, 'r') as fp:
-        config_str = jinja2.Template(fp.read()).render()
+    if hasattr(source, "read"):
+        yamlconf = source.read()
```

Comment:
This change won't work. The parser passes in a filename. A simple if that reads with a context manager if needed will be fine.

Reply:
ack this comment is wrong sorry!